x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java

```diff
@@ -87,25 +91,36 @@ public class NativeRoleMappingStore implements UserRoleMapper {
 
     private static final String ID_PREFIX = DOC_TYPE_ROLE_MAPPING + "_";
 
+    public static final Setting<Boolean> LAST_LOAD_CACHE_ENABLED_SETTING = Setting.boolSetting(
+        "xpack.security.authz.store.role_mappings.last_load_cache.enabled",
+        false,
+        Setting.Property.NodeScope,
+        Setting.Property.Filtered
+    );
```

Comment:
I think it's non-standard behavior, unlike other cache usage we have in ES. There's no TTL and no cache size limit. For our use case we know that the number of role mappings is very small, so it's fine to dump them all into memory; I imagine that's usually the case but there may be customers out there with many mappings, that get more frequent updates/reads, so performance becomes a concern.

If we were to make it a default, non-configurable behavior, we'd also need a more complete strategy around clearing the cache: as you point out, we currently don't clear the cache on all nodes. I don't think it's good security posture to allow a scenario where, e.g., a customer deletes an overly-permissive role mapping, then their cluster goes red, then users with the role mapping that was just deleted continue to access resources (as limited as those may be in a red cluster). For other caches, e.g., the realms cache, a similar scenario is also possible ofc but we make a best effort to clear those. 

Since this was meant to be a targeted quick fix I'd prefer to keep the change as small as possible -- in this case, I think simply _not_ clearing the cache on modification (I can make that change) for now and keeping it configurable, off-by-default seems to be just that. 

As a follow up, I think we can add a cache clearing action, that is called on modification and applies to all nodes. If that's in place, I'd consider exposing this to customers but I think it should remain configurable so that customers with large numbers of role mappings can disable it.  

LMKWYT!